### PR TITLE
fix: upgrade prometheus client_golang. Fixes #1983

### LINKED ISSUE #1983: Import Argo client library but build failed.
# Summary

Tried to utilize client library reference `example/example-golang` to fulfill our need. But build failed. Is something I did wrong? The snippet is from `go.mod`

```go
	github.com/argoproj/argo v2.4.3+incompatible
	github.com/argoproj/pkg v0.0.0-20200102163130-2dd1f3f6b4de // indirect
``` 

# Motivation

At the first, I encountered:
```
# k8s.io/client-go/rest
../../go/pkg/mod/k8s.io/client-go@v11.0.0+incompatible/rest/request.go:598:31: not enough arguments in call to watch.NewStreamWatcher
        have (*versioned.Decoder)
        want (watch.Decoder, watch.Reporter)
```

I googled and follows https://github.com/kubernetes/client-go/issues/584#issuecomment-483214397

Use `go get k8s.io/client-go@master` to upgrade `client-go`

After that, it came with another broken
```
# github.com/argoproj/argo/pkg/client/clientset/versioned/typed/workflow/v1alpha1
../../go/pkg/mod/github.com/argoproj/argo@v2.4.3+incompatible/pkg/client/clientset/versioned/typed/workflow/v1alpha1/workflow_client.go:63:32: undefined: serializer.DirectCodecFactory
# github.com/argoproj/argo/workflow/artifacts/hdfs
../../go/pkg/mod/github.com/argoproj/argo@v2.4.3+incompatible/workflow/artifacts/hdfs/util.go:21:10: options.KerberosClient undefined (type hdfs.ClientOptions has no field or method KerberosClient)
../../go/pkg/mod/github.com/argoproj/argo@v2.4.3+incompatible/workflow/artifacts/hdfs/util.go:22:10: options.KerberosServicePrincipleName undefined (type hdfs.ClientOptions has no field or method KerberosServicePrincipleName)
```

### PR DESCRIPTION
**Pull request by fscnick**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
* [ ] Optional. I've added My organization is added to the README.
* [x] I've signed the CLA and required builds are green. 

Fixes https://github.com/argoproj/argo/issues/1983

### CODE REVIEW

**Comment by simster7** [comment]
Hi @fscnick could you please tell me a bit about why we need to upgrade that library? It's not clear from your issue

**Comment by fscnick** [comment]
If prometheus client with v1.0.0 or higher version is used in a program and also some argo package has been included. That caused an compilation error because the higher version would be taken. However, it has different input arguments. Or it not necessary to upgrade the prometheus client.

**Comment by simster7** [comment]
Will consider this PR after merging #1853

File: workflow/metrics/collector.go [hunk 1]
```diff
@@ -69,7 +69,11 @@ func NewWorkflowRegistry(informer cache.SharedIndexInformer) *prometheus.Registr
 // NewTelemetryRegistry creates a new prometheus registry that collects telemetry
 func NewTelemetryRegistry() *prometheus.Registry {
 	registry := prometheus.NewRegistry()
-	registry.MustRegister(prometheus.NewProcessCollector(os.Getpid(), ""))
+	registry.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{
+		PidFn:        func() (int, error) { return os.Getpid(), nil },
+		Namespace:    "",
+		ReportErrors: true,
+	}))
```
Comment (simster7) [inline workflow/metrics/collector.go]: .

**Comment by simster7** [comment]
@fscnick Seems like we're going to upgrade the client in #2097 along with Go 1.13. I'll be closing this for now, but thanks for submitting this PR.

**Pull request closed without merging**